features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/ImageTypeResolver.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.GradientDrawable
+import androidx.annotation.VisibleForTesting
+
+internal class ImageTypeResolver {
+    fun isDrawablePII(drawable: Drawable, widthInDp: Int, heightInDp: Int): Boolean =
+        drawable !is GradientDrawable &&
```

Comment:
what is special about `GradientDrawable`?

Reply:
GradientDrawable can't be PII because it's just a color gradient, but it receives the dimensions of the view so it would fall foul of this heuristic

Reply:
in theory everything is possible - the most naive counter-example is encoding of some PII information as color. Like QR-code carrying PII, but as Gradient; we don't know what apps can pop up in the market.

Reply:
That's true, on the other hand if we don't exclude this drawable type we may lose the backgrounds of most RippleDrawables since GradientDrawable is at their base